src/Microsoft.OData.Core/UnknownEntitySet.cs

```diff
@@ -44,6 +45,19 @@ public IEdmType Type
             }
         }
 
+        public IEdmEntityType EntityType
+        {
+            get
+            {
+                if (this.entityType == null)
+                {
+                    this.entityType = this.Type?.AsElementType() as IEdmEntityType;

```

Comment:
Is there a possibility that this casting may result in null ?

Reply:
@wachugamaina That's correct. It may result into null and even the previous extension method that was being used in place of this property could return null in some cases. The caller is supposed to anticipate that. I'll mention that it can be null in the property doc.